Makefile

```diff
@@ -145,44 +163,58 @@ $(TOOLBIN):
 
 # Tool Binaries
-CHAINSAW ?= $(TOOLBIN)/chainsaw
-CTLPTL ?= $(TOOLBIN)/ctlptl
+CHAINSAW      ?= $(TOOLBIN)/chainsaw
+CRD_REF_DOCS  ?= $(TOOLBIN)/crd-ref-docs
+CTLPTL        ?= $(TOOLBIN)/ctlptl
 GOLANGCI_LINT ?= $(TOOLBIN)/golangci-lint
-KIND ?= $(TOOLBIN)/kind
-KUSTOMIZE ?= $(TOOLBIN)/kustomize
+KIND          ?= $(TOOLBIN)/kind
+KUSTOMIZE     ?= $(TOOLBIN)/kustomize
+MDBOOK        ?= $(TOOLBIN)/mdbook
 
 # Tool Versions
-CHAINSAW_VERSION ?= $(shell grep 'github.com/kyverno/chainsaw ' ./hack/tools/go.mod | cut -d ' ' -f 2)
-CTLPTL_VERSION ?= $(shell grep 'github.com/tilt-dev/ctlptl ' ./hack/tools/go.mod | cut -d ' ' -f 2)
-GOLANGCI_LINT_VERSION ?= $(shell grep 'github.com/golangci/golangci-lint ' ./hack/tools/go.mod | cut -d ' ' -f 2)
-KIND_VERSION ?= $(shell grep 'sigs.k8s.io/kind ' ./hack/tools/go.mod | cut -d ' ' -f 2)
-KUSTOMIZE_VERSION ?= $(shell grep 'sigs.k8s.io/kustomize/kustomize/v5 ' ./hack/tools/go.mod | cut -d ' ' -f 2)
+CHAINSAW_VERSION      ?= v0.2.11
+CRD_REF_DOCS_VERSION  ?= v0.1.0
+CTLPTL_VERSION        ?= v0.8.35
+GOLANGCI_LINT_VERSION ?= v1.61.0
+KIND_VERSION          ?= v0.24.0
+KUSTOMIZE_VERSION     ?= v5.5.0
+MDBOOK_VERSION        ?= v0.4.45
 
 .PHONY: chainsaw
-chainsaw: $(CHAINSAW)$(CHAINSAW_VERSION)
-$(CHAINSAW)$(CHAINSAW_VERSION): $(TOOLBIN)
+chainsaw: $(CHAINSAW)-$(CHAINSAW_VERSION)
+$(CHAINSAW)-$(CHAINSAW_VERSION): $(TOOLBIN)
 	$(call go-install-tool,$(CHAINSAW),github.com/kyverno/chainsaw,$(CHAINSAW_VERSION))
 
+.PHONY: crd-ref-docs
+crd-ref-docs: $(CRD_REF_DOCS)-$(CRD_REF_DOCS_VERSION)
+$(CRD_REF_DOCS)-$(CRD_REF_DOCS_VERSION): $(TOOLBIN)
+	$(call go-install-tool,$(CRD_REF_DOCS),github.com/elastic/crd-ref-docs,$(CRD_REF_DOCS_VERSION))
+
 .PHONY: ctlptl
-ctlptl: $(CTLPTL)$(CTLPTL_VERSION)
-$(CTLPTL)$(CTLPTL_VERSION): $(TOOLBIN)
+ctlptl: $(CTLPTL)-$(CTLPTL_VERSION)
+$(CTLPTL)-$(CTLPTL_VERSION): $(TOOLBIN)
 	$(call go-install-tool,$(CTLPTL),github.com/tilt-dev/ctlptl/cmd/ctlptl,$(CTLPTL_VERSION))
 
 .PHONY: golangci-lint
-golangci-lint: $(GOLANGCI_LINT)$(GOLANGCI_LINT_VERSION)
-$(GOLANGCI_LINT)$(GOLANGCI_LINT_VERSION): $(TOOLBIN)
+golangci-lint: $(GOLANGCI_LINT)-$(GOLANGCI_LINT_VERSION)
+$(GOLANGCI_LINT)-$(GOLANGCI_LINT_VERSION): $(TOOLBIN)
 	$(call go-install-tool,$(GOLANGCI_LINT),github.com/golangci/golangci-lint/cmd/golangci-lint,$(GOLANGCI_LINT_VERSION))
 
 .PHONY: kind
-kind: $(KIND)$(KIND_VERSION)
-$(KIND)$(KIND_VERSION): $(TOOLBIN)
+kind: $(KIND)-$(KIND_VERSION)
+$(KIND)-$(KIND_VERSION): $(TOOLBIN)
 	$(call go-install-tool,$(KIND),sigs.k8s.io/kind,$(KIND_VERSION))
 
 .PHONY: kustomize
-kustomize: $(KUSTOMIZE)$(KUSTOMIZE_VERSION)
-$(KUSTOMIZE)$(KUSTOMIZE_VERSION): $(TOOLBIN)
+kustomize: $(KUSTOMIZE)-$(KUSTOMIZE_VERSION)
+$(KUSTOMIZE)-$(KUSTOMIZE_VERSION): $(TOOLBIN)
 	$(call go-install-tool,$(KUSTOMIZE),sigs.k8s.io/kustomize/kustomize/v5,$(KUSTOMIZE_VERSION))
 
+.PHONY: mdbook
+mdbook: $(MDBOOK)-$(MDBOOK_VERSION)
+$(MDBOOK)-$(MDBOOK_VERSION): $(TOOLBIN)
+	./hack/mdbook.sh $(MDBOOK) $(MDBOOK_VERSION)
+
```

Comment:
This is under the tools section. Should this be under `generate` instead?

Reply:
No. This is used only to install the mdbook tool. I think we can have an install tool function, or directly put the commands into the Makefile. However, I thought it would provide more clarity to move it into separate script.

Reply:
I see. Maybe a good minor change would be to change the script name to `install-mdbook.sh`. I was assuming that `mdbook.sh` was setting up and invoking the mdbook tool itself -- admittedly partly a problem of me not doing the review all at once and forgetting some context.

Reply:
Fixed.